### PR TITLE
Schema spec for azure site recovery service

### PR DESCRIPTION
**Pull request by mukileshSethu**
Schema spec for azure site recovery service and the respective changes in deployment template.
It also includes the test for schema spec.

**Note:** The swagger spec for azure site recovery is available at [https://github.com/Azure/azure-rest-api-specs/pull/2419]

### CODE REVIEW

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -383,6 +383,16 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-05-01-preview/Microsoft.Sql.json#/resourceDefinitions/servers_keys" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-05-01-preview/Microsoft.Sql.json#/resourceDefinitions/servers_syncAgents" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-05-01-preview/Microsoft.Sql.json#/resourceDefinitions/servers_virtualNetworkRules" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationAlertSettings" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >                   [](start = 172, length = 18)

remove extra spaces.

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 2]
```diff
@@ -383,6 +383,16 @@
...
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationFabrics_replicationProtectionContainers_replicationProtectionContainerMappings" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationFabrics_replicationStorageClassifications_replicationStorageClassificationMappings" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationFabrics_replicationvCenters" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationPolicies" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationRecoveryPlans" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >    				   [](start = 172, length = 10)

remove extra spaces and tabs.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 3]
```diff
@@ -59,6 +59,16 @@
...
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationFabrics_replicationProtectionContainers_replicationProtectionContainerMappings" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationFabrics_replicationStorageClassifications_replicationStorageClassificationMappings" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationFabrics_replicationvCenters" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationPolicies" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationRecoveryPlans" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >                   [](start = 172, length = 18)

remove

File: schemas/2015-01-01/deploymentTemplate.json [hunk 4]
```diff
@@ -59,6 +59,16 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2016-05-15/Microsoft.DevTestLab.json#/resourceDefinitions/labs_schedules" },
                   { "$ref": "https://schema.management.azure.com/schemas/2016-05-15/Microsoft.DevTestLab.json#/resourceDefinitions/labs_virtualmachines" },
                   { "$ref": "https://schema.management.azure.com/schemas/2016-05-15/Microsoft.DevTestLab.json#/resourceDefinitions/labs_virtualnetworks" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-01-10/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_replicationAlertSettings" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >                   [](start = 172, length = 18)

remove

**Comment by Tiano2017** [comment]
@mukileshSethu  please fix the format issue. the rest looks good.

File: schemas/2015-01-01/deploymentTemplate.json (hunk 3 above)
Comment (mukileshSethu) [reply schemas/2015-01-01/deploymentTemplate.json]: format issues right ?

**Pull request merged**